src/isomorphic/devtools/ReactComponentTreeDevtool.js

```diff
@@ -106,6 +106,11 @@ var ReactComponentTreeDevtool = {
   },
 
   purgeUnmountedComponents() {
+    if (ReactComponentTreeDevtool._preventPurging) {
```

Comment:
Could you add another devtool for the test that checks the state at an appropriate time? If not or if that's too clunky, I think this seems fine too.